benchmarks/Sentry.Benchmarks/LastActiveSpanBenchmarks.cs

```diff
@@ -0,0 +1,54 @@
+using BenchmarkDotNet.Attributes;
+using Perfolizer.Mathematics.Randomization;
+
+namespace Sentry.Benchmarks;
+
+public class LastActiveSpanBenchmarks
+{
+    private const string Operation = "Operation";
+    private const string Name = "Name";
+
+    [Params(1, 10, 100)]
+    public int SpanCount;
+
+    private IDisposable _sdk;
+
+    [GlobalSetup(Target = nameof(CreateScopedSpans))]
+    public void EnabledSdk() => _sdk = SentrySdk.Init(Constants.ValidDsn);
+
+    [GlobalCleanup(Target = nameof(CreateScopedSpans))]
+    public void DisableDsk() => _sdk.Dispose();
+
+    [Benchmark(Description = "Create spans for scope access")]
+    public void CreateScopedSpans()
+    {
+        var rnd = new Random();
+        var transaction = SentrySdk.StartTransaction(Name, Operation);
+        SentrySdk.WithScope(scope =>
+        {
+            scope.Transaction = transaction;
+            for (var i = 0; i < SpanCount; i++)
+            {
+                // Simulates a scenario where TransactionTracer.GetLastActiveSpan will be called frequently
+                // See: https://github.com/getsentry/sentry-dotnet/blob/c2a31b4ead03da388c2db7fe07f290354aa51b9d/src/Sentry/Scope.cs#L567C1-L567C68
+                var span = transaction.StartChild(Operation);
+                if (rnd.Next(2) < 1)
```

Comment:
Ideally we don't use random in the benchmark. So we can get consistent values for each run

Reply:
Good call. I got rid of the "sometimes test `GetLastActiveSpan` and sometimes don't" logic from the benchmark entirely... The impact of this optimization will depend on how frequently customers' applications call `GetLastActiveSpan` but we can't really know (or similate) the likelihood of that in a benchmark... so currently our benchmarks give us something clsoe to the maximum possible performance gain a customer could get, if their code was structured exactly like our benchmark (pretty slim chance of that).